nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -1252,6 +1262,55 @@ def exec(self, cmd_line_flag=False, layer_name=None):
         return QueryAnswer(True, self.name1 + ' and ' + self.name2 + ' are semantically equivalent.',
                            numerical_result=0)
 
+    def check_equivalence_optimized(self, layer_name=None):
+        conn_props1 = self.config1.allowed_connections_optimized(layer_name)
+        conn_props2 = self.config2.allowed_connections_optimized(layer_name)
+        all_conns1, all_conns2 = self.filter_conns_by_input_or_internal_constraints(conn_props1.all_allowed_conns,
+                                                                                    conn_props2.all_allowed_conns)
+        if all_conns1 != all_conns2:
```

Comment:
might be simpler to read if checking equality first, so that not most of the code with this indentation.

Reply:
Done